ucs/database.cpp

```diff
@@ -280,16 +304,136 @@ bool Database::LoadChatChannels()
 	}
 
 	for (auto row = results.begin(); row != results.end(); ++row) {
-		std::string channelName     = row[0];
-		std::string channelOwner    = row[1];
+		std::string channelName = row[0];
+		std::string channelOwner = row[1];
 		std::string channelPassword = row[2];
 
-		ChannelList->CreateChannel(channelName, channelOwner, channelPassword, true, atoi(row[3]));
-	}
 
+		if (IsOnChannelBlockList(channelName)) {
+			if (!(channelOwner == "*Block*")) { // Dont show that channels were blocked when the block entries themselves are detected.
+				LogDebug("Blocked channel [{}] load from database due to being blocked by server operator.", channelName);
+			}
+		}
+		else {
+			ChannelList->CreateChannel(channelName, channelOwner, channelPassword, true, atoi(row[3]), false);
+		}
+	}
 	return true;
 }
 
+bool Database::IsOnChannelBlockList(std::string channelName) {
+	// Ignore the input if it is empty
+	if (channelName == "") {
+		return false;
+	}
+	// Convert channelName to lowercase for case-insensitive comparison
+	std::string channelNameLower = channelName;
+	std::transform(channelNameLower.begin(), channelNameLower.end(), channelNameLower.begin(), ::tolower);
+	// Check if channelName is already in the BlockedChannelNames vector
+	bool isFound = false;
+	for (const std::string& name : BlockedChannelNames) {
```

Comment:
`Strings::Contains` has a string vector method that be used to simplify this

```cpp
bool Strings::Contains(std::vector<std::string> container, std::string element)
{
	return std::find(container.begin(), container.end(), element) != container.end();
}
```

Reply:
Updated